### PR TITLE
feat: support use of module keyword

### LINKED ISSUE #150: In grammar, allow `module` keyword

### PR DESCRIPTION
**Pull request by ewanharris**
## Description

Adds support for the `module` keyword in the grammar, there's a couple things that I'm not 100% certain of (hence draft for feedback):

* Passing both `module` and `model` causes an error, but I don't know if we can improve the error message received - `no viable alternative at input '\nmodule'`
* Currently validation fails because when parsing the DSL we don't know the schema version to use so will error, I think we might need to allow passing the version obtained from a `go.mod` file into validate in some way? Which is probably outside the scope of this PR.
* I changed the existing validation to throw if we can't determine a schema version, otherwise when we get to the switch further down we can't blow up trying to determine line numbers, this might want to be changed to only throw in the case of a single file module.

## References

Closes #150, #151

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Comment by rhamzeh** [comment]
We shouldn't be validating a partial model, only after all the pieces are collected can we validate

File: tests/data/dsl-semantic-validation-cases.yaml [hunk 1]
```diff
@@ -1153,6 +1153,21 @@
         symbol: "allowed_ip"
         errorType: condition-not-used
 
+- name: supports module keyword
```
Comment (rhamzeh) [inline tests/data/dsl-semantic-validation-cases.yaml:1156]: Yeah this is correctly failing, but the name is off.

**Comment by ewanharris** [comment]
@rhamzeh I'm not sure I follow, are you saying that when processing modular models the intention is to construct the final model and then pass that to `validateDSL` rather than the individual model modules?

**Comment by rhamzeh** [comment]
> when processing modular models the intention is to construct the final model and then pass that to `validateDSL` rather than the individual model modules?

Yes, the parser will handle validating the model syntactically, but you cannot validate it semantically, because it will need data from the other files.

E.g. user is not in that file, so validation will naturally fail until you have collected all the files in the model

**Comment by ewanharris** [comment]
So should `validateDSL` throw an error along the lines of `cannot parse individual module` if passed a module DSL?

**Review by rhamzeh — DISMISSED** [review]

**Pull request merged**